exchanges/refocus/src/refocusExchange.ts

```diff
@@ -1,12 +1,36 @@
 import { pipe, tap } from 'wonka';
 import { Exchange, Operation } from '@urql/core';
 
+// Source: https://github.com/vercel/swr/blob/4d6be3df441f925d235a4b26914efcc85ae335a3/src/libs/is-document-visible.ts
+const isDocumentVisible = (): boolean => {
+  if (
+    typeof document !== 'undefined' &&
+    typeof document.visibilityState !== 'undefined'
+  ) {
+    return document.visibilityState !== 'hidden';
+  }
+  // always assume it's visible
+  return true;
+};
+
+// Source: https://github.com/vercel/swr/blob/4d6be3df441f925d235a4b26914efcc85ae335a3/src/libs/is-online.ts
+const isOnline = (): boolean => {
+  if (typeof navigator.onLine !== 'undefined') {
+    return navigator.onLine;
+  }
+  // always assume it's online
+  return true;
+};
+
 export const refocusExchange = (): Exchange => {
   return ({ client, forward }) => ops$ => {
     const watchedOperations = new Map<number, Operation>();
     const observedOperations = new Map<number, number>();
 
-    window.addEventListener('focus', () => {
+    window.addEventListener('visibilitychange', () => {
+      if (!isDocumentVisible() || !isOnline()) {
```

Comment:
Let's just inline `isDocumentVisible`. There's also not really any reason to enter the source, since it's a common API with just a one liner, so this can just become:

```suggestion
      if (typeof document !== 'undefined' && document.visibilityState === 'hidden') {
```

Reply:
> Thanks for picking this up! First on a small side-note, we document our changes for our automatically generated changelogs using "changeset" (that's what the comment above is about). Just check these instructions here on how to document a change here and probably mark this one as "minor": https://github.com/FormidableLabs/urql/blob/main/CONTRIBUTING.md#how-do-i-document-a-change-for-the-changelog

Sorry, I should have read the contributing doc before submitting my PR 😅

Reply:
Shouldn't it be: `document.visibilityState === 'visible'` (and not 'hidden') ?

Reply:
I don't think so? You made this `return` so it's the reverse, if we're hidden we don't re-issue the operation. Maybe it's more intuitive to just wrap all following statements in this if-statement instead to avoid this confusion (then it'd be `typeof document !== 'object' || document.visibilityState === 'visible'`

Reply:
Oh yes, indeed I forgot I made an early return here. Thanks :)